packages/diagnostic/src/diagnose.ts

```diff
@@ -0,0 +1,136 @@
+import ts from 'typescript';
+
+import { ESLint } from 'eslint';
+import { DiagnosticService } from './DiagnosticService';
+
+export async function diagnose(basePath: string, configName = 'tsconfig.json'): Promise<boolean> {
+  try {
+    const configFile = ts.findConfigFile(basePath, ts.sys.fileExists, configName);
+
+    if (!configFile) {
+      console.log(`Config file ${configName} not found in ${basePath}`);
+      return false;
+    }
+
+    const service = new DiagnosticService(basePath, configFile);
+
+    for (const sourceFile of service.getRootSourceFiles()) {
+      let diagnostics: ts.Diagnostic[] = [];
+
+      console.log('', '', sourceFile.fileName, '', '');
+
+      // TODO: Remove
+      diagnostics = service.getSemanticDiagnostics(sourceFile);
+      for (const diagnostic of diagnostics) {
+        console.log(
+          formatDiagnosticMessage(diagnostic),
+          ts.getLineAndCharacterOfPosition(sourceFile, diagnostic.start!),
+          ''
+        );
+      }
+
+      await lint(sourceFile);
+      service.updateSourceFile(sourceFile);
+
+      diagnostics = service.getSemanticDiagnostics(sourceFile);
+      for (const diagnostic of diagnostics) {
+        // TODO: Add comments
+        console.log(
+          formatDiagnosticMessage(diagnostic),
+          ts.getLineAndCharacterOfPosition(sourceFile, diagnostic.start!),
+          ''
+        );
+      }
+
+      await lint(sourceFile);
+      service.updateSourceFile(sourceFile);
+    }
+  } catch (e) {
+    console.error('Diagnose failed', e);
+    return false;
+  }
+
+  return true;
+}
+
+/*
```

Comment:
any code commented out could you specify why its commented out and not just removed?

Reply:
in progress... will be removed